components/feature/downloads/src/test/java/mozilla/components/feature/downloads/share/ShareDownloadMiddlewareTest.kt

```diff
@@ -0,0 +1,275 @@
+import org.mockito.Mockito.verify
+import java.io.File
+import java.nio.charset.StandardCharsets
+
+@RunWith(AndroidJUnit4::class)
+class ShareDownloadMiddlewareTest {
+    private val testDispatcher = TestCoroutineDispatcher()
+
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule(testDispatcher)
+
+    @Test
+    fun `ShareDownloadMiddleware will intercept DownloadAction#AddDownloadAction`() {
+        val shareMiddleware = spy(ShareDownloadMiddleware(mock(), mock()))
+        doNothing().`when`(shareMiddleware).startSharing(any())
+        val fakeMiddleware = spy(FakeMiddleware())
+        val store = BrowserStore(middleware = listOf(
+            shareMiddleware,
+            fakeMiddleware
+        ))
+        val download = DownloadState("testDownload", source = DownloadState.Source.SHARE)
+        val action = DownloadAction.AddDownloadAction(download)
+
+        store.dispatch(action).joinBlocking()
+        testDispatcher.advanceUntilIdle()
+
+        verify(shareMiddleware).startSharing(download)
+        verify(fakeMiddleware, never()).invoke(any(), any(), eq(action))
+    }
+
+    @Test
+    fun `ShareDownloadMiddleware will pass forward actions other than DownloadAction#AddDownloadAction`() {
+        val shareMiddleware = spy(ShareDownloadMiddleware(mock(), mock()))
+        doNothing().`when`(shareMiddleware).startSharing(any())
+        val fakeMiddleware = spy(FakeMiddleware())
+        val store = BrowserStore(middleware = listOf(
+            shareMiddleware,
+            fakeMiddleware
+        ))
+        // AddDownloadAction but with DownloadState.Source.DOWNLOAD should not be intercepted
+        val download = DownloadState("testDownload", source = DownloadState.Source.DOWNLOAD)
+        val addDownloadAction = DownloadAction.AddDownloadAction(download)
+        val removeDownloadAction = DownloadAction.RemoveDownloadAction("")
+        val removeAllDownloadsAction = DownloadAction.RemoveAllDownloadsAction
+        val updateDownloadAction = DownloadAction.UpdateDownloadAction(mock())
+        val restoreDownloadsAction = DownloadAction.RestoreDownloadsStateAction
+        val restoreDownloadAction = DownloadAction.RestoreDownloadStateAction(mock())
+
+        // dispatch all 6 DownloadAction
+        store.dispatch(addDownloadAction).joinBlocking()
+        store.dispatch(removeAllDownloadsAction).joinBlocking()
+        store.dispatch(removeDownloadAction).joinBlocking()
+        store.dispatch(updateDownloadAction).joinBlocking()
+        store.dispatch(restoreDownloadAction).joinBlocking()
+        store.dispatch(restoreDownloadsAction).joinBlocking()
+        testDispatcher.advanceUntilIdle()
+
+        verify(shareMiddleware, never()).startSharing(any())
+        // 6 actions dispatched from above + the default InitAction
+        verify(fakeMiddleware, times(7)).invoke(any(), any(), any())
+    }
+
+    @Test
+    @Ignore("Call not recorded: \"zero interactions with this mock.\"")
```

Comment:
Why we are commit this ignore test?

Reply:
I looked at this as something that can be rechecked when we upgrade the mockito version and would help more knowing that we should retry testing this instead of maybe not knowing it is not tested.
But this seems superfluous. Will remove.